include/alpaka/rand/Philox/PhiloxSingle.hpp

```diff
@@ -98,6 +109,15 @@ namespace alpaka
                     state.position = static_cast<decltype(state.position)>(state.position + (offset & 3));
                     offset += state.position < 4 ? 0 : 4;
                     state.position -= state.position < 4 ? 0 : 4u;
+                    while(state.position > 0)
+                    {
+                        // Shift state results to allow hard coded access to element zero.
+                        // This will avoid high register usage on NVIDIA devices.
+                        state.result[0] = state.result[1];
+                        state.result[1] = state.result[2];
+                        state.result[2] = state.result[3];
+                        --state.position;
```

Comment:
This is wrong, we need to remember the old position and decrease the value for the loop only.